### PR TITLE
Add dG errors by component, plot to RBFE results

### PR DESCRIPTION
**Pull request by mcwitt**
* Adds new fields to `rbfe.SimulationResult`:
  - `dG_errs_by_lambda_by_component`: array of bootstrapped BAR errors for each energy component and pair of adjacent lambda windows
  - `dG_errs_png`: plot analogous to `overlap_summary_png` for bootstrapped BAR errors. Example:

### CODE REVIEW

File: timemachine/fe/rbfe.py [hunk 1]
```diff
@@ -424,27 +463,22 @@ def estimate_free_energy_given_initial_states(initial_states, protocol, temperat
     ukln_by_lambda_by_component = np.array(ukln_by_component_by_lambda).swapaxes(0, 1)
 
     lambdas = [s.lamb for s in initial_states]
-
-    _, (ax_top, ax_btm) = plt.subplots(2, 1, figsize=(7, 9))
     overlaps_by_lambda = np.array([pair_overlap_from_ukln(u_kln) for u_kln in ukln_by_lambda_by_component.sum(axis=0)])
-    plot_overlap_summary(ax_top, ["Overall"], lambdas, [overlaps_by_lambda])
-
+    dG_errs_by_lambda_by_component = np.array(
+        [[dG_err_from_ukln(u_kln) for u_kln in ukln_by_lambda] for ukln_by_lambda in ukln_by_lambda_by_component]
```
Comment (proteneer) [inline timemachine/fe/rbfe.py:468]: is this in reduced units still? (not sure where the conversion to kJ/mol is happening)

Comment (mcwitt) [reply timemachine/fe/rbfe.py]: Fixed in 6710e48

**Review by proteneer — APPROVED** [review]
lgtm

**Review by maxentile — APPROVED** [review]
Looks good!

I continue to have some generic confusion how to interpret these per-energy-component diagnostics (whether per-component overlap as currently reported, or per-component dG_err as proposed). dG_err may be preferable to overlap in some respects since it is a quantity with interpretable units. (Curiosity: Are there examples where the interpretations differ for diagnostics based on dG_err vs. overlap?)

In practice these per-component decomposition are helpfully suggestive of which component to focus attention on, even though in principle (1) neither the estimate nor the error admit a decomposition of this form, (2) this logic could yield arbitrarily bad "false-positive" diagnoses in principle.

A possible short-term enhancement may be to modify these plots with specialized depiction for windows suffering from MD instability. (to preempt questions of form: "was the large error / low overlap due to instability or just inefficiency")

Another possible short-term enhancement may be to add diagnostics for when specific assumptions are violated. (E.g. if a harmonic angle term is being interpolated in a way that assumes the angle marginal is approximately Gaussian, possibly add a diagnostic to flag windows where that feature is highly non-Gaussian.)

A possible long-term enhancement may be to develop a more principled attribution diagnostic (details hazy, but perhaps based on `<du_dparams>_params` and `dparams_dlam`...)

**Review by jkausrelay — APPROVED** [review]
lgtm

**Pull request merged**